### PR TITLE
Fixes #22969: Rewrite angular app "ComplianceMode" in Elm

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://issues.rudder.io/issues/22969

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/DataTypes.elm [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module ComplianceMode.DataTypes exposing (..)
+
+import Http exposing (Error)
+import Http.Detailed
+
+--
+-- All our data types
+--
+
+type ComplianceMode = FullCompliance | ChangesOnly | ReportsDisabled | UnknownMode
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/DataTypes.elm:10]: I don 't think you need an UnknownMode. It should make an error if the value is invalid.

File: webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/Init.elm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+init : { contextPath : String, hasWriteRights : Bool, complianceMode : {name : String, heartbeatPeriod : Int}, globalMode : {name : String, heartbeatPeriod : Int} } -> ( Model, Cmd Msg )
+init flags =
+  let
+    getComplianceMode : String -> ComplianceMode
+    getComplianceMode mode =
+      case mode of
+        "full-compliance"  -> FullCompliance
+        "changes-only"     -> ChangesOnly
+        "reports-disabled" -> ReportsDisabled
+        _                  -> UnknownMode
```
Comment (VinceMacBuche) [inline webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/Init.elm:27]: This should either be an error, not ignored

**Comment by RaphaelGauthier** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/DataTypes.elm (hunk 1 above)
Comment (RaphaelGauthier) [reply webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/DataTypes.elm]: I've renamed UnknownMode to ErrorMode, but I've kept it to make it easier to handle this error case, by displaying a message in the form, preventing it from being saved and displaying an error notification.

Comment (VinceMacBuche) [reply webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/DataTypes.elm]: GReat !

**Review by VinceMacBuche — APPROVED** [review]

File: webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/Init.elm (hunk 2 above)
Comment (RaphaelGauthier) [reply webapp/sources/rudder/rudder-web/src/main/elm/sources/ComplianceMode/Init.elm]: I used this "Error" mode to display an error in the form, but the user can still select a new valid report mode and save his changes.

**Comment by RaphaelGauthier** [comment]
OK, merging this PR

**Pull request merged**